### ISSUE TITLE
[Bug]: collection:xxxxxxxxx or partition:[] not loaded into memory when search

### ISSUE THREAD

**Issue by clk1st** [milvus-io/milvus#24472]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.2.6
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2):milvus-sdk-java,2.2.4
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 8核64G
- GPU: 0
- Others:
```

### Current Behavior

Call loadCollection, the execution result is displayed as follows:
2023-05-29 9:41:04.747 [search-center,,,] 48105 --- [io-61156-exec-1] i.m.client.AbstractMilvusGrpcClient : LoadCollectionParam{collectionName='xxxxxxxxx', syncLoad=true', syncLoadWaitingInterval =2000', syncLoadWaitingTimeout=30', replicaNumber=1'}
2023-05-29 9:41:35.742 [search-center,,,] 48105 --- [io-61156-exec-1] i.m.client.AbstractMilvusGrpcClient : Waiting load thread is timeout, loading process may not be finished
2023-05-29 9:41:35.750 [search-center,,,] 48105 --- [io-61156-exec-1] c.j.s.b.s.s.impl.CollectionServiceImpl : Collection synchronously loaded memory successfully R{status=0, data=RpcStatus {msg='Success'}}

But after a long time of execution, the result of calling search still shows:
2023-05-29 10:40:59.411 [search-center,,,] 48105 --- [o-61156-exec-10] i.m.client.AbstractMilvusGrpcClient : SearchParam{collectionName='xxxxxxxxx', partitionNames='[]' , metricType=IP, target vectors count=1, vectorFieldName='seg_vector', topK=20, nq=1, expr='user_id == 234001180599084', params='{"nprobe":80}'}
2023-05-29 10:40:59.562 [search-center,,,] 48105 --- [o-61156-exec-10] i.m.client.AbstractMilvusGrpcClient : SearchRequest failed:
collection:xxxxxxxxx or partition:[] not loaded into memory when search
2023-05-29 10:40:59.563 [search-center,,,] 48105 --- [o-61156-exec-10] c.j.s.b.s.service.impl.DataServiceImpl : Search error: collection:xxxxxxxxx or partition:[] not loaded into memory when searching

**Comment by yanliang567** [comment]
@clk1st sounds like a known issue that fixed in v2.2.8. could you please retry with v2.2.8?
/assign @clk1st 
/unassign

**Comment by clk1st** [comment]
{
	"level": "ERROR",
	"time": "2023/05/29 04:39:31.370 +00:00",
	"caller": "querynode/segment_loader.go:126",
	"message": "load failed, OOM if loaded",
	"collectionID": 440352734423146465,
	"segmentType": "Sealed",
	"loadSegmentRequest msgID": 4,
	"error": "InsufficientMemoryToLoad, load segment failed, OOM if load, collectionID = 440352734423146465, maxSegmentSize = 1449 MB, concurrency = 1, usedMemAfterLoad = 56123 MB, currentAvailableMemCount = 20681 MB, currentTotalMem = 58200 MB, thresholdFactor = 0.900000",
	"stack": "github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:126\ngithub.com/milvus-io/milvus/internal/querynode.(*loadSegmentsTask).Execute\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/load_segment_task.go:81\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:109\ngithub.com/milvus-io/milvus/internal/querynode.(*taskScheduler).taskLoop\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/task_scheduler.go:131"
}
